### ISSUE TITLE
Add task to check if a PR is for a library with bundled types

### ISSUE THREAD

**Issue by MysteryBlokHed** [DefinitelyTyped/DefinitelyTyped#67361]
Every now and then there's a PR to add types for a library which already bundles its own types. It'd probably be worth having some check, maybe through @typescript-bot, to see if a library contains `.d.ts` files and to add a comment for the PR author.

This would also mean that reviewers don't have to worry about not noticing that a library already exports its own types, then merging a `@types` package that won't be used.

**Comment by jakebailey** [comment]
Surprisingly, it's kinda difficult to determine if a package actually is typed or not. They may mention types in their `package.json` which is a definite indicator, but some packages use fallback resolution where the `d.ts` files live next to the implementation. For https://jakebailey.dev/dt-dashboard/ I have made a heuristic to try and make things work, though.

Annoyingly, npm has a `TS` indicator but no API to retrieve that info or info on how it actually works.

@andrewbranch Is ATTW able to determine if a package is already typed?

**Comment by andrewbranch** [comment]
Yep. It should be easy to add that step to dtslint since we’re already going to fetch the implementation package at the beginning of the run.

**Comment by jakebailey** [comment]
ATTW running is now a part of CI.

**Comment by andrewbranch** [comment]
Honestly I’m not sure what it will do the way I have it configured. I think attw will just report what happens from the user’s perspective when both the typed package and the superfluous types are installed. I probably need to add a separate check for this to make it fail.
